docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+# Smart Contracts Basics
+
+A smart contract is a program that runs on the blockchain. It allows you to add functionality to an account ranging from simple ones like a todo application to a fully-fledged RPG game which runs entirely on the blockchain.
+
+This guide will show you how to develope a basic EOS smart contract with **DUNE** and with the C++ programming language. Also the basic smart contract concepts will be explained along the way.
+
+## Create Contract and Test Accounts
+
+To deploy a smart contract you need an account to deploy it to. Create an account `hello` with the following command:
+
+```shell
+dune --create-account hello
+```
+
+Create a second account, `ama`, for test purposes.
+
+```shell
+dune --create-account ama
+```
+
+## Create a DUNE Smart Contract Application
+
+Make sure you have DUNE installed. Otherwise follow the [DUNE development setup](../10_getting-started/10_dune-guide/index.md#installation) documentation.
+
+There are two types of DUNE smart contract applications.
+
+- [The bare application](#create-build-and-deploy-a-dune-bare-application)
+- [The cmake application](#create-build-and-deploy-a-dune-cmake-application)
+
+Pick any of the above and create a DUNE application which will be used for the entire guide forward.
+
+### Create, Build and Deploy a DUNE Bare Application
+
+The bare application contains just the minimum files needed to develop, compile and deploy one smart contract to the blockchain. Run the following command to create a bare `hello` application.
+
+```shell
+dune --create-bare-app hello .
+cd hello
+ls
+```
+
+The result of this command is a `hello` directory with four files in it:
+
+- README.txt, a text file which contains information about how to build this application.
+- hello.contracts.md, the ricardian contract markdown file.
+- hello.hpp, the C++ header file.
+- hello.cpp, the C++ implementation file.
+
+#### Build the DUNE Bare Application
+
+To build the DUNE bare application run the following command:
+
+```shell
+dune -- cdt-cpp /host/<LOCAL-PATH-TO-HELLO-PARENT>/hello/hello.cpp -o /host/<LOCAL-PATH-TO-HELLO-PARENT>/hello/hello.wasm
+```
+
+The above command you can run it from directory on your local machine, just make sure that the `<LOCAL-PATH-TO-HELLO-PARENT>` is replace with the corresponding path of the parent directory of the hello folder.
```

Comment:
The command itself is pretty self-explanatory without this second explanation. No need for it.